src/GUI/RevEng.Core/ReplacingCandidateNamingService.cs

```diff
@@ -42,28 +43,32 @@ public override string GenerateCandidateIdentifier(DatabaseTable originalTable)
 
                 return candidateStringBuilder.ToString();
             }
-            else if (schema.Tables.Count > 0)
-            {
-                var newTableName = _customNameOptions
-                    .Where(s => s.SchemaName == schema.SchemaName)
-                    .SelectMany(t => t.Tables.Where(n => n.Name == originalTable.Name))
-                    .FirstOrDefault()?.NewName;
 
-                if (string.IsNullOrWhiteSpace(newTableName))
-                {
-                    candidateStringBuilder.Append(ToPascalCase(originalTable.Name));
+            string newTableName = string.Empty;
 
-                    return candidateStringBuilder.ToString();
-                }
-
-                candidateStringBuilder.Append(newTableName);
-
-                return candidateStringBuilder.ToString();
+            if (schema.Tables.Any(t => t.Name == originalTable.Name))
+            {
+                newTableName = schema.Tables.FirstOrDefault(t => t.Name == originalTable.Name)?.NewName;
+            }
+            else if (!string.IsNullOrEmpty(schema.TableRegexPattern) && schema.TablePatternReplaceWith != null)
+            {
+                newTableName = Regex.Replace(originalTable.Name, schema.TableRegexPattern, schema.TablePatternReplaceWith);
```

Comment:
Please add options with a timeout

Reply:
Did add a timeout to regex replace method